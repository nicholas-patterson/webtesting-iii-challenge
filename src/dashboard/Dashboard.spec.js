import React from "react";
import renderer from "react-test-renderer";
import Dashboard from "./Dashboard";
import { render, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

describe("<Dashboard/>", () => {
  // 2. write this test
  it("matches snapshot", () => {
    const tree = renderer.create(<Dashboard />); // generates a DOM tree

    // snapshots are a JSON representation of the DOM tree
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it("it shows props text when event clicked", () => {
    const comp = render(<Dashboard />);
    console.log(comp);
  });
});

describe("<Controls/>", () => {
  it("button text changes to reflect the state of the door", () => {
    const { getByTestId } = render(<Dashboard />);
    const toggleLocked = getByTestId("togglelocked"); // Lock Gate
    const toggleOpen = getByTestId("toggleopen"); // Close Gate
    console.log(toggleOpen.textContent);
    fireEvent.click(toggleOpen);
    expect(toggleOpen).toHaveTextContent("Open Gate");
    fireEvent.click(toggleLocked);
    expect(toggleLocked).toHaveTextContent("Unlock Gate");
  });
  it("the closed toggle button is disabled if the gate is locked", () => {
    const { getByTestId } = render(<Dashboard />);
    const toggleLocked = getByTestId("togglelocked"); // Lock Gate
    const toggleOpen = getByTestId("toggleopen"); // Close Gate
    fireEvent.click(toggleOpen);
    fireEvent.click(toggleLocked);
    expect(toggleOpen).toBeDisabled();
  });
  it("the locked toggle button is disabled if the gate is open", () => {
    const { getByTestId } = render(<Dashboard />);
    const toggleLocked = getByTestId("togglelocked"); // Lock Gate
    const toggleOpen = getByTestId("toggleopen"); // Close Gate
    expect(toggleLocked).toBeDisabled();
  });
});
