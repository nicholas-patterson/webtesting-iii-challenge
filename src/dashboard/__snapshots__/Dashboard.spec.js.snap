// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard/> matches snapshot 1`] = `
Array [
  <div
    className="display panel"
  >
    <div
      className="led green-led"
      data-testid="lockeddiv"
    >
      Unlocked
    </div>
    <div
      className="led green-led"
      data-testid="closeddiv"
    >
      Open
    </div>
  </div>,
  <div
    className="controls panel"
  >
    <button
      className="toggle-btn"
      data-testid="togglelocked"
      disabled={true}
      onClick={[Function]}
    >
      Lock Gate
    </button>
    <button
      className="toggle-btn"
      data-testid="toggleopen"
      disabled={false}
      onClick={[Function]}
    >
      Close Gate
    </button>
  </div>,
]
`;
